### PR TITLE
[DROOLS-1025] Incremental Rule Refactor + new tests

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

File: drools-compiler/src/test/java/org/drools/compiler/integrationtests/Misc2Test.java [hunk 1]
```diff
@@ -8309,4 +8310,44 @@ public void testIn() {
         assertEquals( "Bob", list.get( 0 ) );
     }
 
+    @Test
+    public void testXXX() {
```
Comment (triceo) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/Misc2Test.java:8314]: Could we maybe have a more descriptive name? :-)

Comment (manstis) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/Misc2Test.java:8314]: I can always help with descriptive test names; I have quite a reputation ;-)

Comment (mariofusco) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/Misc2Test.java:8314]: This PR is incomplete, Mark and me are still working on it. I used that test only to compare the behaviour of drools with the same set of rules but with and without incremental compilation. I'll remove that test when I'll be done with this.

**Comment by mariofusco** [comment]
Merged by https://github.com/droolsjbpm/drools/commit/8d52a6a34

**Pull request closed without merging**